src/main/java/com/puppycrawl/tools/checkstyle/checks/imports/ImportOrderCheck.java

```diff
@@ -401,6 +401,10 @@ else if (option == ImportOrderOption.BOTTOM) {
             }
             doVisitToken(ident, isStatic, isLastImportAndNonStatic);
 
+            if (isLastImportAndNonStatic) {
```

Comment:
Fix was added to option `bottom`. Does `top` have a similar issue as their code looks similar?

Reply:
@djydewang , please recheck top mode, if issue exists we need register it, if fix is complicated to do now.
I will be ok to merge even without TOP fix, as issue was for bottom only.